eventsources/eventing.go

```diff
@@ -565,5 +565,5 @@ func filterEvent(data []byte, filter *v1alpha1.EventSourceFilter) (bool, error)
 		params[strings.ReplaceAll(key, "-", "_")] = value
 	}
 	env := expr.GetFuncMap(params)
-	return expr.EvalBool(strings.ReplaceAll(filter.Expression, "-", "_"), env)
+	return expr.EvalBool(filter.Expression, env)
```

Comment:
and a test?

Reply:
Added a test case for EvalBool for expressions with `-` and added output for example for original issue to PR.